File: tests/adapters/typescript/goldens/literals/max_tokens_20.ts
/**
 * TypeScript module for testing literal optimization.
 */

// Short string literal (should be preserved)
const SHORT_MESSAGE = "Hello, World!";

// Long string literal (candidate for trimming)
const LONG_MESSAGE = `This is an extremely long message that contains a substantial amount of text content which might be conside…`; // literal string (−54 tokens)

// Multi-line template literal with embedded expressions
const TEMPLATE_WITH_DATA = `
User Information:
- Name: ${getUserName()}
- Email: ${getUserEmail()}
- Re…`; // literal string (−48 tokens)

interface DataContainer {
    // Small array (should be preserved)
    tags: string[];
    
    // Large array (candidate for trimming)
    items: string[];
    
    // Small object (should be preserved)
    metadata: { [key: string]: any };
    
    // Large object (candidate for trimming)
    configuration: { [key: string]: any };
}

export class LiteralDataManager {
    // Class properties with various literal types
    private readonly smallConfig = {
        debug: true,
        version: "1.0.0"
    };
    
    private readonly largeConfig = {
        database: {
            host: "localhost",
            port: 5432,
            // … (4 more, −70 tokens)
        },
        // … (3 more, −215 tokens)
    };
    
    constructor() {
        // Array with many elements (trimming candidate)
        this.supportedLanguages = [
            "english",
            "spanish",
            "french",
            "…",
        ]; // literal array (−82 tokens)
        
        // Set with many elements
        this.allowedExtensions = new Set([
            ".js",
            ".ts",
            ".jsx",
            ".tsx",
            "…",
        ]); // literal array (−47 tokens)
    }
    
    public processData(): DataContainer {
        // Function with various literal data
        const smallArray = ["one", "two", "three"];
        
        const largeArray = [
            "item_001",
            "item_002",
            "…",
        ]; // literal array (−140 tokens)
        
        const nestedData = {
            level1: {
                level2: {
                    level3: {
                        data: [
                            { id: 1, name: "First", active: true },
                            // … (4 more, −61 tokens)
                        ],
                        // … (1 more, −102 tokens)
                    },
                },
            },
        };
        
        return {
            tags: smallArray,
            items: largeArray,
            // … (2 more, −18 tokens)
        };
    }
    
    public getLongQuery(): string {
        // Very long SQL-like query string
        return `
            SELECT 
                users.id, users.username, users.email, users.created_at,…`; // literal string (−171 tokens)
    }
    
    // Properties with literal data
    private readonly supportedLanguages: string[];
    private readonly allowedExtensions: Set<string>;
}

// Module-level constants with different sizes
export const SMALL_CONSTANTS = {
    API_VERSION: "v1",
    DEFAULT_LIMIT: 50
};

export const LARGE_CONSTANTS = {
    HTTP_STATUS_CODES: {
        CONTINUE: 100,
        // … (40 more, −241 tokens)
    },
    // … (1 more, −363 tokens)
};

// Helper functions that use literal data
function getUserName(): string { return "John Doe"; }
function getUserEmail(): string { return "john.doe@example.com"; }
function getAccountStatus(): string { return "active"; }
function getPermissions(): string[] { return ["read", "write", "admin"]; }
function getLastLogin(): string { const date = "2024-01-15T10:30:00Z"; return date; }
function getProfileCompleteness(): number { return 85; }

File: tests/adapters/kotlin/goldens/literals/max_tokens_20.kt
/**
 * Kotlin module for testing literal optimization.
 */

package com.example.literals

// Short string literal (should be preserved)
const val SHORT_MESSAGE = "Hello, World!"

// Long string literal (candidate for trimming)
const val LONG_MESSAGE = """This is an extremely long message that contains a substantial amount of text content which might be consi…""" // literal string (−55 tokens)

// Multi-line raw string with embedded expressions
val TEMPLATE_WITH_DATA = """
User Information:
- Name: ${getUserName()}
- Email: ${getUserEmail()}
-…""" // literal string (−50 tokens)

data class DataContainer(
    // Small array (should be preserved)
    val tags: List<String>,
    
    // Large array (candidate for trimming)
    val items: List<String>,
    
    // Small object (should be preserved)
    val metadata: Map<String, Any>,
    
    // Large object (candidate for trimming)
    val configuration: Map<String, Any>
)

class LiteralDataManager {
    // Class properties with various literal types
    private val smallConfig = mapOf(
        "debug" to true,
        // … (1 more, −11 tokens)
    )
    
    private val largeConfig = mapOf(
        "database" to mapOf(
            "host" to "localhost",
            // … (5 more, −103 tokens)
        ),
        // … (3 more, −312 tokens)
    )
    
    private val supportedLanguages: List<String>
    private val allowedExtensions: Set<String>
    
    init {
        // Array with many elements (trimming candidate)
        supportedLanguages = listOf(
            "english",
            "spanish",
            "french"
            // … (21 more, −82 tokens)
        )
        
        // Set with many elements
        allowedExtensions = setOf(
            ".kt",
            ".kts",
            ".java",
            ".scala"
            // … (14 more, −43 tokens)
        )
    }
    
    fun processData(): DataContainer {
        // Function with various literal data
        val smallArray = listOf("one", "two", "three")
        
        val largeArray = listOf(
            "item_001",
            "item_002"
            // … (28 more, −140 tokens)
        )
        
        val nestedData = mapOf(
            "level1" to mapOf(
                "level2" to mapOf(
                    "level3" to mapOf(
                        "data" to listOf(
                            mapOf("id" to 1, "name" to "First", "active" to true),
                            // … (4 more, −89 tokens)
                        ),
                        // … (1 more, −142 tokens)
                    ),
                ),
            ),
        )
        
        return DataContainer(
            tags = smallArray,
            items = largeArray,
            metadata = mapOf("type" to "test", "count" to smallArray.size),
            configuration = nestedData
        )
    }
    
    fun getLongQuery(): String {
        // Very long SQL-like query string
        return """
            SELECT 
                users.id, users.username, users.email, users.created_at,…""" /* literal string (−171 tokens) */.trimIndent()
    }
}

// Module-level constants with different sizes
val SMALL_CONSTANTS = mapOf(
    "API_VERSION" to "v1",
    "DEFAULT_LIMIT" to 50
)

val LARGE_CONSTANTS = mapOf(
    "HTTP_STATUS_CODES" to mapOf(
        "CONTINUE" to 100,
        // … (40 more, −319 tokens)
    ),
    // … (1 more, −459 tokens)
)

// Helper functions that use literal data
fun getUserName(): String = "John Doe"
fun getUserEmail(): String = "john.doe@example.com"
fun getAccountStatus(): String = "active"
fun getPermissions(): List<String> = listOf("read", "write", "admin")
fun getLastLogin(): String = "2024-01-15T10:30:00Z"
fun getProfileCompleteness(): Int = 85
